### ISSUE TITLE
`L071 Unnecessary trailing whitespace`

### ISSUE THREAD

**Issue by Nate-Wessel** [sqlfluff/sqlfluff#4441]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

A problem is being reported under the wrong rule number. I have `L001` configured as a warning only, and there are indeed a few instances of trailing whitespace in the code below. Oh well. The problem is that one (only one) of these trailing whitespaces is incorrectly reported under rule `L071` , though still with the correct ''unnecessary trailing whitespace" message. Because `L071` isn't configured as a warning, it reports as an error and the linting fails when it shouldn't. 

### Expected Behaviour

The linting should pass with warnings only. 

### Observed Behaviour

The linting fails because a rule violation is reported under the wrong rule. 

### How to reproduce

Run `sqlfluff lint` on the following file:

```sql
CREATE TABLE data_requests.i0418_transit_tt_geom AS (

    WITH node_lookup(uid, corridor, from_street, to_street, direction, start_node, end_node) AS (
        VALUES
        ---------- PARLIAMENT -----------  
        -- Dundas St to Queen St
        (51, 'Parliament St', 'Dundas St', 'Queen St', 'SB', 30416368, 30416331),
        (52, 'Parliament St', 'Queen St', 'Dundas St', 'NB', 30416331, 30416368),
        -- Queen St to Front St
        (53, 'Parliament St', 'Queen St', 'Front St', 'SB', 30416331, 30416252),
        (54, 'Parliament St', 'Front St', 'Queen St', 'NB', 30416252, 30416331) 
    ),

    routing AS (
        SELECT
            nl.uid,
            nl.corridor,
            nl.from_street,
            nl.to_street,
            nl.direction,
            nl.start_node,
            nl.end_node,
            segs.segment_list,
            segs.length AS corridor_length,
            array_length(segs.segment_list, 1) AS num_seg
        FROM node_lookup AS nl,
            LATERAL(SELECT * FROM congestion.get_segments_btwn_nodes(nl.start_node::int, nl.end_node::int)) AS segs
    ),

    unnest_cte AS (
        SELECT
            rgs.uid,
            rgs.corridor,
            rgs.from_street,
            rgs.to_street,
            rgs.direction,
            rgs.start_node,
            rgs.end_node,
            rgs.segment_list,
            rgs.num_seg,
            rgs.corridor_length,
            unnest(rgs.segment_list) AS segment_id
        FROM routing AS rgs
    )

    SELECT
        uc.uid,
        uc.corridor,
        uc.from_street,
        uc.to_street,
        uc.direction,
        uc.start_node,
        uc.end_node,
        uc.num_seg,
        uc.segment_id,
        cns.total_length AS seg_length, 
        cns.geom
    FROM unnest_cte AS uc
    INNER JOIN congestion.network_segments AS cns
        ON cns.segment_id = uc.segment_id

                                                                                                           
)


```

I get the following message:

```
FAIL
L:  11 | P:  80 | L001 | WARNING: Unnecessary trailing whitespace.
L:  11 | P:  80 | L071 | Unnecessary trailing whitespace.
L:  27 | P: 116 | L016 | WARNING: Line is too long.
L:  56 | P:  40 | L001 | WARNING: Unnecessary trailing whitespace.
L:  56 | P:  40 | L008 | WARNING: Unnecessary trailing whitespace.
L:  62 | P:   1 | L001 | WARNING: Unnecessary trailing whitespace.
L:  62 | P: 108 | L016 | WARNING: Line is too long.
All Finished 📜 🎉!
```

### Dialect

postgres

### Version

1.4.5

### Configuration

```
[sqlfluff]
dialect = postgres
exclude_rules = L009, L030, L031, L032, L051, L055
warnings = L001, L005, L006, L008, L016, L023, L050

[sqlfluff:rules]
tab_space_size = 4
indent_unit = space
max_line_length = 100
aliasing = explicit

[sqlfluff:rules:L010]
capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
With the latest `sqlfluff` I see proper WARNING for all trailing whitespaces:
```bash
 sqlfluff lint test.sql -d postgres 
== [test.sql] FAIL
L:   3 | P:   5 | LT05 | Line is too long (97 > 80). [layout.long_lines]
L:   3 | P:  21 | LT01 | WARNING: Expected single whitespace between naked
                       | identifier and start bracket '('.
                       | [layout.spacing]
L:   7 | P:   9 | LT05 | Line is too long (81 > 80). [layout.long_lines]
L:   8 | P:   9 | LT05 | Line is too long (81 > 80). [layout.long_lines]
L:  11 | P:  80 | LT01 | WARNING: Unnecessary trailing whitespace.
                       | [layout.spacing]
L:  27 | P:  13 | LT05 | Line is too long (115 > 80).
                       | [layout.long_lines]
L:  27 | P:  20 | LT01 | WARNING: Expected single whitespace between 'LATERAL'
                       | keyword and start bracket '('. [layout.spacing]
L:  56 | P:  40 | LT01 | WARNING: Unnecessary trailing whitespace.
                       | [layout.spacing]
L:  60 | P:  12 | ST09 | Joins should list the table referenced earlier first.
                       | [structure.join_condition_order]
L:  62 | P:   1 | LT01 | WARNING: Unnecessary trailing whitespace.
                       | [layout.spacing]
L:  64 | P:   1 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
All Finished 📜 🎉!
```
